docs/reference/garbage-collection.md

```diff
@@ -0,0 +1,88 @@
+The GC job does not remove any commits: you will still be able to use commits containing hard-deleted objects,
+but trying to read these objects from lakeFS will result in a `409 Gone` HTTP status.
+
+## Understanding Garbage Collection
+
+For every branch, the GC job retains deleted objects for the number of days defined for the branch.
+In the absence of a branch-specific rule, the default rule for the repository is used.
+If an object is present in more than one branch ancestry, it is retained according to the rule with the largest number of days between those branches.
+That is, it is hard-deleted only after the retention period has ended for all relevant branches.
+
+**Note:** In order for an object to be hard-deleted, it must be deleted from all branches.
+You should delete stale branches to prevent them from retaining old objects.
+{: .note }
+
+Example GC rules for a repository:
+```json
+{
+  "default_retention_days": 21,
+  "branches": [
+    {"branch_id": "main", "retention_days": 28},
+    {"branch_id": "dev", "retention_days": 7}
+  ]
+}
+```
+
+In the above example, objects are retained for 21 days after deletion by default. However, if they are present in the main `branch`, they are retained for 28 days.
+Objects present in the `dev` branch (but not in the `main` branch), are retained for 7 days after they are deleted.
+
+## Configuring GC rules
+
+Use the `lakectl` CLI to define the GC rules: 
+
+```bash
+cat <<EOT >> example_repo_gc_rules.json
+{
+  "default_retention_days": 21,
+  "branches": [
+    {"branch_id": "main", "retention_days": 28},
+    {"branch_id": "dev", "retention_days": 7}
+  ]
+}
+EOT
+
+lakectl gc set-config lakefs://example-repo -f example_repo_gc_rules.json 
+```
+
+## Running the GC job
+
+```bash
+spark-submit --class io.treeverse.clients.GarbageCollector \
+  -c spark.hadoop.lakefs.api.url=https://lakefs.example.com:8000/api/v1  \
+  -c spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY> \
+  -c spark.hadoop.lakefs.api.secret_key=<LAKEFS_SECRET_KEY> \
+  -c spark.hadoop.fs.s3a.access.key=<S3_ACCESS_KEY> \
+  -c spark.hadoop.fs.s3a.secret.key=<S3_SECRET_KEY> \
+  --packages io.lakefs:lakefs-spark-client-301_2.12:0.1.0 \
+  example-repo us-east-1
+```
+## Considerations
+1. lakeFS will never delete objects outside your repository's storage namespace.
+   In particular, objects that were imported using `lakefs import` or `lakectl ingest` will not be affected by GC jobs.
+   
+1. In cases where deleted objects are brought back to life while a GC job is running, said objects may or may not be
+   deleted. Such actions include:
+   1. Reverting a commit in which a file was deleted.
+   1. Branching out from an old commit.
+   1. Expanding the retention period of a branch.
+   1. Creating a branch from an existing branch, where the new branch has a longer retention period.
+
+1. You should not run more than a single GC job at the same time.
```

Comment:
What happens if I do?  Can we help me somehow avoid this?

Reply:
Thought it over, no issue with running two of them concurrently, if you really want it. Removing this bullet